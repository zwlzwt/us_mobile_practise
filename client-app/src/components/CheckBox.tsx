import React from 'react';
import styled from 'styled-components'

const CheckInput = styled.div`
  display: flex;
  padding-top: 20px;
  align-items: center;
  margin-top: 18px;
  border-top: 1px solid rgba(25, 35, 50, 0.1);
  input {
    width: 20px;
    height: 20px;
  }
`

const CheckPolicy = styled.p`
  margin-left: 15px;
  display:inline-block;
  font-style: normal;
  font-weight: normal;
  font-size: 20px;
  line-height: 32px;

  letter-spacing: -0.4px;

  color: #192332;
   @media screen and (max-width: 414px) {
     font-size: 18px;
     margin-left: 10px;
   }
`

interface Props {
  checked: boolean,
  onChange: Function
}


const CheckBox = ({
  checked,
  onChange
}: Props) => {
  return (
    <CheckInput>
      <input type='checkbox' checked={checked} onChange={() => onChange()}/>
      <CheckPolicy>Agree to the Terms â„‡ Privacy policy</CheckPolicy>
    </CheckInput>
  )
}

export default CheckBox