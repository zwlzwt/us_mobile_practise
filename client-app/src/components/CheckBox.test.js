import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CheckBox from './CheckBox';


it('renders correctly react-test-renderer', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<CheckBox />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});