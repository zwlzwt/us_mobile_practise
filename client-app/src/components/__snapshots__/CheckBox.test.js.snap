// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `
<ForwardRef(styled.div)>
  <input
    onChange={[Function]}
    type="checkbox"
  />
  <ForwardRef(styled.p)>
    Agree to the Terms â„‡ Privacy policy
  </ForwardRef(styled.p)>
</ForwardRef(styled.div)>
`;
